rusk-wallet/src/lib/cache.rs

```diff
@@ -0,0 +1,96 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+
+use dusk_bytes::{DeserializableSlice, Serializable};
+use phoenix_core::Note;
+use rusqlite::{params, Connection};
+
+pub(crate) fn open(data_dir: &PathBuf) -> Connection {
+    let db_path = Path::new(data_dir).join("cache.db");
+
+    let cache = Connection::open(db_path).unwrap();
```

Comment:
Do we really want to leave this `unwrap()` call here? There's been quite a bit of effort in error handling. I think we could be more explicit or at least propagate the error back and implement the corresponding `From` trait in the `crate::Error`